### ISSUE TITLE
Start failed: BlockNotPinnedError: Block is not pinned

### ISSUE THREAD

**Issue by mutantcornholio** [polkadot-api/polkadot-api#352]
While trying to repoduce #350, I started to get this error during startup:

```
Start failed: BlockNotPinnedError: Block is not pinned
    at Observable._subscribe (/Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/@polkadot-api/client/src/observableClient/chainHead/chainHead.ts:100:28)
    at Observable._trySubscribe (/Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/Observable.ts:244:19)
    at /Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/Observable.ts:234:18
    at Object.errorContext (/Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/util/errorContext.ts:29:5)
    at Observable.subscribe (/Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/Observable.ts:220:5)
    at /Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/operators/share.ts:239:27
    at OperatorSubscriber.<anonymous> (/Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/util/lift.ts:24:18)
    at /Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/Observable.ts:226:22
    at Object.errorContext (/Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/util/errorContext.ts:29:5)
    at Observable.subscribe (/Users/cornholio/work/parity/polkadot-testnet-faucet/node_modules/rxjs/src/internal/Observable.ts:220:5)
```
Here's the log:
[not_pinned.log](https://github.com/polkadot-api/polkadot-api/files/14779141/not_pinned.log)

**Comment by josepot** [comment]
Could you please also share with us the code that you used to trigger this error? Because with the logs and the code, we will then be able to reproduce exactly what happened and create integration tests to prevent these bugs from occurring in the future.

Also, I have not looked into the logs yet. However, please be aware that we only keep as pinned blocks those blocks that are:
- either above or equal to the currently finalized block
- those blocks that despite being "old" (prior to the currently finalized block), the consumer is still running queries against.

Although, I'm pretty sure that this is must be a bug caused by a change in #332, which assumes that certain blocks won't be seen by the consumer, when it seems that's not actually the case.

Anyways, thanks a lot for reporting this and for providing the logs. If you could please also share the code that you used for triggering this issue, that would be even better. Thanks again!

**Comment by mutantcornholio** [comment]
Well, it's the same code and the same situation as in here: https://github.com/polkadot-api/polkadot-api/issues/350#issue-2210894146
Same crash at the startup, but a different stacktrace

**Comment by josepot** [comment]
Also fixed by #354 thanks!
